### ISSUE TITLE
jq error userdata.json:0

### ISSUE THREAD

**Issue by ghost** [jp1995/kappa-launcher#18]
Configuration file found, proceeding
jq: error (at /home/mads/.config/kpl/userdata.json:0): Cannot iterate over null (null)
json file successfully populated
jq: error (at /home/mads/.config/kpl/followdata.json:0): Cannot iterate over null (null)
Using monospaced font found kpl config
jq: error (at /home/mads/.config/kpl/followdata.json:0): Cannot iterate over null (null)
jq: error (at /home/mads/.config/kpl/followdata.json:0): Cannot iterate over null (null)

**Comment by ghost** [comment]
I am on Arch Linux with Xfce 4.18

**Comment by jp1995** [comment]
Can you post the contents of your userdata and maybe followdata files after you get this? The followdata file might just be empty, or it might have a new error that I'm not dealing with right now.

**Comment by ghost** [comment]
Sure,
followdata: `{"error":"Bad Request","status":400,"message":"Missing required parameter \"user_id\""}`
userdata: `{"error":"Unauthorized","status":401,"message":"Invalid OAuth token"}`

**Comment by jp1995** [comment]
Have you generated your OAuth key and placed it into the config file? It's possible it has expired, too, if it was working for you previously.

It's certainly not supposed to report success in the output there so I'll look into that.

*Edit:* Yeah, it's strange because when I insert a typo into my OAuth token, it gives me the correct error. There's also a separate error for when the token is missing entirely, which does not appear to be the case for you. I'd try regenerating the token.

**Comment by ghost** [comment]
Alright will do tomorrow, the token is as old as this issue though. But I will definitely try it out. 

Stupid Question but does the token need to be created on the same OS? Because my current token was created on a Windows machine since I can not login to Twitch on Arch, because of Twitch privacy restrictions no browser is working for me. So Kappa Launcher is also the current best alternative for a smooth experience.

**Comment by ghost** [comment]
Alright, had some irl stuff to do, so I could only just do it. I replaced the OAuth token with a new one, but the same issue persists. I tried to write a faulty OAuth key on purpose but that just kept the same error, but instead of 3 times, just 1 time.

**Comment by jp1995** [comment]
Sorry for the late reply, missed your comment. Have you done a full config wipe and reinstall yet? I'm quite perplexed about this issue you're getting, as I can't really replicate it.

**Comment by ghost** [comment]
Yes I did, I also completely clean built and deleted the config folder beforehand. Is there any Rust configuration you have?

**Comment by ghost** [comment]
sorry, I meant rofi not rust

**Comment by ghost** [comment]
It's an jq error, I am not an expert but I will try looking into it myself, maybe [this](https://stackoverflow.com/questions/54794749/jq-error-at-stdin0-cannot-iterate-over-null-null) helps

**Comment by ghost** [comment]
I manged to fix it myself by adding 

Check if userdata.json exists and is not empty
```
if [ ! -s "$MAIN_PATH/userdata.json" ]; then
    echo "ERROR: userdata.json is empty or does not exist."
    exit 1
fi
```

Check if followdata.json exists and is not empty
```
if [ ! -s "$MAIN_PATH/followdata.json" ]; then
    echo "ERROR: followdata.json is empty or does not exist."
    exit 1
fi
```

Check if userdata.json contains valid data
```
USER_ID=$(jq -r '.data[].id' $MAIN_PATH/userdata.json)
if [ -z "$USER_ID" ]; then
    echo "ERROR: userdata.json does not contain valid data."
    exit 1
fi
```

Check if followdata.json contains valid data
```
if ! jq -e '.data | length > 0' $MAIN_PATH/followdata.json >/dev/null; then
    echo "ERROR: followdata.json does not contain valid data."
    exit 1
fi 
```

Between the populate check and the call for names of live channels, then delete followdata.json and userdata.json and let them repopulate.

**Comment by ahmetkotanci** [comment]
> I manged to fix it myself by adding
> 
> Check if userdata.json exists and is not empty
> 
> ```
> if [ ! -s "$MAIN_PATH/userdata.json" ]; then
>     echo "ERROR: userdata.json is empty or does not exist."
>     exit 1
> fi
> ```
> 
> Check if followdata.json exists and is not empty
> 
> ```
> if [ ! -s "$MAIN_PATH/followdata.json" ]; then
>     echo "ERROR: followdata.json is empty or does not exist."
>     exit 1
> fi
> ```
> 
> Check if userdata.json contains valid data
> 
> ```
> USER_ID=$(jq -r '.data[].id' $MAIN_PATH/userdata.json)
> if [ -z "$USER_ID" ]; then
>     echo "ERROR: userdata.json does not contain valid data."
>     exit 1
> fi
> ```
> 
> Check if followdata.json contains valid data
> 
> ```
> if ! jq -e '.data | length > 0' $MAIN_PATH/followdata.json >/dev/null; then
>     echo "ERROR: followdata.json does not contain valid data."
>     exit 1
> fi 
> ```
> 
> Between the populate check and the call for names of live channels, then delete followdata.json and userdata.json and let them repopulate.

hi where did you add these. Im getting exact same error
